icu4c/source/tools/genbrk/genbrk.cpp

```diff
@@ -234,7 +234,10 @@ int  main(int argc, char **argv) {
     if (U_FAILURE(status)) {
         exit(status);
     }
-    if(encoding!=nullptr ){
+    if (encoding == nullptr) {
+        // In the absence of a BOM, assume the rule file is in UTF-8.
+        encoding = "UTF-8";
```

Comment:
So this is a change in behavior of this tool.
In a follow-up change, we should document (genbrk usage output, maybe User Guide) that genbrk looks for a Unicode signature byte sequence and otherwise assumes UTF-8.

https://unicode.org/faq/utf_bom.html
https://www.unicode.org/glossary/#unicode_signature